specification/compiler/model/metamodel.ts

```diff
@@ -120,21 +120,39 @@ export class Property {
   type: ValueOf
   required: boolean
   description?: string
-  annotations?: Record<string, string>
+  /**
+   * If specified takes precedence over `name` when generating code. `name` is always the value
+   * to be sent over the wire
+   */
+  alternateName?: string[]
+  /** An optional set of aliases for `name` */
+  aliases?: string[]
 }
 
 // ------------------------------------------------------------------------------------------------
 // Type definitions
 
 export type TypeDefinition = Interface | Request | Union | Enum | TypeAlias
 
+export type Depreciable = TypeDefinition | EnumMember
+export const isDepreciable = (type: object): type is Depreciable => {
+  return (type as Depreciable).name !== undefined;
```

Comment:
Isn't `name` catching a too big nest of objects that aren't Depreciable (and same for `BehaviorAttachable`? Also we should avoid adding code in `metamodel.ts`.

Reply:
It is, all these typeguards are not even needed. Have updated the code to remove these